File: 240304/최고의 33위치/best-place-of-33.py
n = int(input())
arr = [list(map(int,input().split()))for _ in range(n)]
ans =0
for row in range(n):
    for col in range(n):
        matrix=0
        if row+3 > n or col+3> n :
            continue
        for i in range(row,row+3):
            for j in range(col,col+3):
                matrix += arr[i][j]
        ans = max(ans,matrix)
print(ans)

File: 240304/최고의 33위치/best-place-of-33.kt
fun main(){
    val N = readLine()!!.toInt()

    val arr = MutableList<List<Int>>(0){List(0){0}}

    for(i in 0 until N){
        arr.add(readLine()!!.trimEnd().split(" ").map{it.toInt()})
    }

    var max = 0
    for(r in 0 until N-2){
        for(c in 0 until N-2){
            var tempSum = 0
            for(y in r until r+3){
                for(x in c until c+3){
                    tempSum += arr[y][x]
                }
            }

            max = maxOf(tempSum, max)
        }
    }

    print(max)
}